dags/benk/generate.py

```diff
@@ -31,6 +33,9 @@
                 Relate(definition.catalog, collection.collection, relation) for relation in collection.relations
             ]
 
+        # Start with initialising / migrating before imports
+        chain(*initialise.get_leaf_nodes(), flatten_list([i.get_start_nodes() for i in imports]))
```

Comment:
This doesn't seem right. You're unpacking the leaf nodes, so all leaf nodes will be chained one after the other. What you want is start the imports after all leaf nodes have ended separately.

Say initialise has two leaf nodes, L1 and L2, with import i1, i2 and i3. What you do now is:

L1 -> L2 -> [i1, i2, i3], or written out:

L1 -> L2
L2 -> i1
L2 -> i2
L2 -> i3

What you want is [L1, L2] -> [i1, i2, i3], or, written out:

L1 -> i1
L1 -> i2
L1 -> i3
L2 -> i1
L2 -> i2
L2 -> i3

Of course, when there is only 1 leaf node, it doesn't really matter in this case. See what happens when you leave out the rules starting with L1 above. The result is the same. It goes wrong when there are multiple leaf nodes, and it's always possible that a new leaf node will be added in the future. Then the script will break.

Instead, use
`cross_downstream(initialise.get_leaf_nodes(), flatten_list([i.get_start_nodes() for i in imports]))`

Reply:
fixed